ports/stm32/pyb_can.c

```diff
@@ -162,6 +196,32 @@ STATIC mp_obj_t pyb_can_init_helper(pyb_can_obj_t *self, size_t n_args, const mp
     // set the CAN configuration values
     memset(&self->can, 0, sizeof(self->can));
 
+    // Calculate CAN bit timing from baudrate if provided
+    if (args[ARG_baudrate].u_int != 0) {
+        uint32_t baudrate = args[ARG_baudrate].u_int;
+        uint32_t sampoint = args[ARG_sample_point].u_int;
+        uint32_t can_kern_clk = pyb_can_get_source_freq();
+        bool timing_found = false;
+
+        // The following max values work on all MCUs for classical CAN.
+        for (int brp = 1; brp < 512 && timing_found == false; brp++) {
+            for (int bs1 = 1; bs1 < 16 && timing_found == false; bs1++) {
+                for (int bs2 = 1; bs2 < 8 && timing_found == false; bs2++) {
+                    if ((baudrate == (can_kern_clk / (brp * (1 + bs1 + bs2)))) &&
+                        ((sampoint * 10) == (((1 + bs1) * 1000) / (1 + bs1 + bs2)))) {
+                        args[ARG_bs1].u_int = bs1;
+                        args[ARG_bs2].u_int = bs2;
+                        args[ARG_prescaler].u_int = brp;
+                        timing_found = true;
+                    }
+                }
+            }
+        }
+        if (timing_found == false) {
+            mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Invalid baudrate and/or sampling point!"));
```

Comment:
please use lowercase letter to start, and no exclamation mark, eg "couldn't match baudrate and sample point"